tests/test_fastavro.py

```diff
@@ -1723,3 +1723,53 @@ def test_appending_records_different_schema_fails(tmpdir):
             fastavro.writer(new_file, different_schema, [{"field": 1}])
 
         assert "does not match file writer_schema" in str(exc)
+
+
+def test_appendable_raises_valuerror(tmpdir):
+    """six.appendable() raises ValueError when file is only 'a' mode."""
+    test_file = str(tmpdir.join("test.avro"))
+
+    with open(test_file, "a") as new_file:
+        new_file.write('this phrase forwards cursor position beyond zero')
+        with pytest.raises(ValueError) as exc:
+            appendable(new_file)
+        assert "you must use the 'a+' mode" in str(exc)
+
+
+def test_appendable_true_nonzero(tmpdir):
+    """six.appendable() returns false when file_like.tell() is non-zero."""
+    test_file = str(tmpdir.join("test.avro"))
+
+    with open(test_file, "a+b") as new_file:
+        new_file.write(b'this phrase forwards cursor position beyond zero')
+        assert appendable(new_file)
+
+
+def test_appendable_false_zero(tmpdir):
+    """six.appendable() returns false when file_like.tell() returns 0."""
+    class MockFileLike:
+        def seekable(self):
+            return True
+
+        def tell(self):
+            # mock a 0 position
+            return 0
+
+    assert not appendable(MockFileLike())
+
+
+def test_appendable_false_unseekable_stream():
+    """File streams that cannot seek return False."""
+
+    class MockFileLike:
```

Comment:
Can you change this name to `MockStreamLike` as that seems to better describe it.